Minio.Functional.Tests/FunctionalTest.cs

```diff
@@ -4775,6 +4775,138 @@ internal static async Task GetObject_Test2(IMinioClient minio)
+    {
+        var startTime = DateTime.Now;
+        var bucketName = GetRandomName(15);
+        var objectName = GetRandomObjectName(10);
+        string contentType = null;
+        var tempFileName = "tempFile-" + GetRandomName();
+        var args = new Dictionary<string, string>
+            (StringComparer.Ordinal)
+            {
+                { "bucketName", bucketName }, { "objectName", objectName }, { "contentType", contentType }
+            };
+        try
+        {
+            await Setup_Test(minio, bucketName).ConfigureAwait(false);
+            // Don't Put the object, so we can hit "ObjectNotFound" exception
+            Stream strm;
+            await using ((strm = rsg.GenerateStreamFromSeed(1 * MB)).ConfigureAwait(false))
+            {
+                var file_write_size = strm.Length;
+                long file_read_size = 0;
+                var getObjectArgs = new GetObjectArgs()
+                    .WithBucket(bucketName)
+                    .WithObject(objectName)
+                    .WithCallbackStream(async (stream, cancellationToken) =>
+                    {
+                        var fileStream = File.Create(tempFileName);
+
+                        await stream.CopyToAsync(fileStream, cancellationToken).ConfigureAwait(false);
+                        await fileStream.DisposeAsync().ConfigureAwait(false);
+
+                        var writtenInfo = new FileInfo(tempFileName);
+                        file_read_size = writtenInfo.Length;
+
+                        Assert.AreEqual(file_write_size, file_read_size);
+                        File.Delete(tempFileName);
+                    });
+                _ = await minio.GetObjectAsync(getObjectArgs).ConfigureAwait(false);
+            }
+
+            new MintLogger("GetObjectNegObjNotFound_Test3", getObjectSignature,
+                "Tests whether GetObjectAsync hits ObjectNotFoundException",
+                TestStatus.FAIL, DateTime.Now - startTime, "Failed to hit ObjectNotFound exception", args: args).Log();
+            throw new Exception("Failed to hit ObjectNotFound exception");
+        }
+        catch (ObjectNotFoundException)
+        {
+            new MintLogger("GetObjectNegObjNotFound_Test3", getObjectSignature,
+                "Tests whether GetObjectAsync hits ObjectNotFoundException",
+                TestStatus.PASS, DateTime.Now - startTime, args: args).Log();
+        }
+        catch (Exception ex)
+        {
+            new MintLogger("GetObjectNegObjNotFound_Test3", getObjectSignature,
+                "Tests whether GetObjectAsync hits ObjectNotFoundException",
+                TestStatus.FAIL, DateTime.Now - startTime, ex.Message, ex.ToString(), args: args).Log();
+            throw;
+        }
+        finally
+        {
+            if (File.Exists(tempFileName))
+                File.Delete(tempFileName);
+            await TearDown(minio, bucketName).ConfigureAwait(false);
+        }
```

Comment:
There is no need to create a stream, because there shouldn't be an object that can read. So just raise an exception if it does hit the callback-stream. You can also use a `Stopwatch` in .NET to track call durations.

This reduces code a lot:
```suggestion
        var stopwatch = Stopwatch.StartNew();
        var bucketName = GetRandomName(15);
        var objectName = GetRandomObjectName(10);
        var args = new Dictionary<string, string>
            (StringComparer.Ordinal)
            {
                { "bucketName", bucketName },
                { "objectName", objectName },
            };
        try
        {
            await Setup_Test(minio, bucketName).ConfigureAwait(false);
            // Don't Put the object, so we can hit "ObjectNotFound" exception
            var getObjectArgs = new GetObjectArgs()
                .WithBucket(bucketName)
                .WithObject(objectName)
                .WithCallbackStream(_ => throw new Exception("Should never be reached"));
            _ = await minio.GetObjectAsync(getObjectArgs).ConfigureAwait(false);

            new MintLogger("GetObjectNegObjNotFound_Test3", getObjectSignature,
                "Tests whether GetObjectAsync hits ObjectNotFoundException",
                TestStatus.FAIL, stopwatch.Elapsed, "Failed to hit ObjectNotFound exception", args: args).Log();
            throw new Exception("Failed to hit ObjectNotFound exception");
        }
        catch (ObjectNotFoundException)
        {
            new MintLogger("GetObjectNegObjNotFound_Test3", getObjectSignature,
                "Tests whether GetObjectAsync hits ObjectNotFoundException",
                TestStatus.PASS, stopwatch.Elapsed, args: args).Log();
        }
        catch (Exception ex)
        {
            new MintLogger("GetObjectNegObjNotFound_Test3", getObjectSignature,
                "Tests whether GetObjectAsync hits ObjectNotFoundException",
                TestStatus.FAIL, stopwatch.Elapsed, ex.Message, ex.ToString(), args: args).Log();
            throw;
        }
        finally
        {
            await TearDown(minio, bucketName).ConfigureAwait(false);
        }
```

Reply:
Done